### DISCUSSION TITLE
ACG banned in some countries?

### DISCUSSION THREAD

**Discussion by RedshiftDrift** [a-cosmology-group/acg#109] [General]
ACG member Виталий Петриков tells me that he cannot register to GitHub from Russia.  The reason is not clear since he has access to the website (on a GitHub IP) and to `Discussion` (also on GitHub).
I just realized that internet wars have resulted in GitHub IP addresses being blocked (see [https://en.wikipedia.org/wiki/Censorship_of_GitHub#Russia](https://en.wikipedia.org/wiki/Censorship_of_GitHub#Russia)), but it's difficult to know the details.

This may also explain why we have not heard from A. Ghosh, A. Barenbaum, and others.

Since the [cosmology.info](https://cosmology.info/) website is now on a GitHub IP address, the website and discussion forum may not be accessible to some internet users around the world.

**Very disappointing**

Edit: GitHub and cosmology.info is currently (2023-7-20) accessible through the `Chinese Firewall`, and also accessible to some Russian users.

**Comment by ftessier** [comment]
Oh my! Disappointing indeed. Since the website is build statically, I guess it can be built on github and then served elsewhere. But who knows what other hosting company will ban access?

**Reply by RedshiftDrift** [reply]
I doubt `GitHub Pages` can send the website to another IP...

**Reply by ftessier** [reply]
No I mean you can redirect your `cosmology.info` to a third-party web hosting service (accessible from everywhere), and simply copy the github directory to that site, say automatically, daily, whatever.

**Reply by RedshiftDrift** [reply]
Do you know how to do that?

**Reply by RedshiftDrift** [reply]
A website mirror would work, but this doesn't solve the lack of access to `Discussions`...

**Reply by ftessier** [reply]
I know, `Discussions` are the issue. For the website hosting, I presume that any web hosting service would do, you would just copy the content of `acg/` there?

**Reply by ftessier** [reply]
Something like https://whc.ca/fr, but there are tons of web hosting services of course: https://ca.cybernews.com/lp/best-web-hosting/. Today the challenge is to find one where you can just dump your files...

**Reply by RedshiftDrift** [reply]
The content of `acg/` is in markdown format, it's not a website.  One would need to get the website itself with a similar command you used to copy `https://cosmology,info` to your computer ;-) and ftp that to the HostMonster provider.

**Reply by ftessier** [reply]
https://support.hostinger.com/ has [github integration](https://support.hostinger.com/en/articles/1583302-how-to-deploy-a-git-repository), meaning (if I understand correctly) that it uses a web hook to update your files on hostinger when you make a change on github. Looking for services that allow that would make it much easier than having to copy things over SSH manually or with a script.

**Reply by RedshiftDrift** [reply]
ACG still has an account on HostMonster, it was paid up to Feb. 2024.

**Reply by ftessier** [reply]
Markdown is not an issue: it is Jekyll that converts markdown to html when building the static site.

**Reply by ftessier** [reply]
If you want, we can test that. I can generate the website locally with Jekyll, and package the files to dump on HostMonster, to see if that works. If there is no github integration, it is straighforward to write a small script which will check for changes regularly, build the website and dump it on the hosting platform automatically (or semi-automatically!).

Sigh, after all the work to point `cosmology.info` to github! github would remain crucial for development.

**Reply by ftessier** [reply]
Jekyll traces its origin to a blog website builder, so it may be possible to leverage its capabilities in this regard to maintain a blog with comments to host the discussions as well, integrated in the same cosmology.info website.

**Comment by ftessier** [comment]
Re Discussions: I just asked ChatGPT to write a small web application to create a discussion forum similar to github Discussions. It is pretty good! It would not be too hard to integrate a small forum into the existing website, I reckon.

**Comment by RedshiftDrift** [comment]
> If you want, we can test that...

@ftessier all of the above suggestions are possible, but I have exceeded my time quota for ACG.  I had hoped that the move to GitHub would lessen my workload, but the last two months have been more work than the former `ACG 2.0 website & forum`.

Now I want to work on the science exclusively, so I won't be able to do these tests with you.

Concerning the future of ACG, there may be a [*Plan Continuation Bias*](https://en.wikipedia.org/wiki/Sunk_cost#Plan_continuation_bias) that is becoming a problem (I learned this as a student pilot 😉!)

**Reasons to move ACG to GitHub:**
1. allows members to write equations in `Discussion`,
2. allows members to modify the webpage and comments in a Git environment,
3. produces a nicer looking webpage,
4. will save the cost of the web host and forum,
5. will save maintenance time, because members can edit the website,
6. a script can be written to archive `Discussion`.

**Actual outcomes after a month:**
1. only 1 member makes extensive use of equations,
2. only 1 member added a paper on the website, no discussion happened,
3. *website 3.0* gets an average of 2.5 clicks in Google search/day, **same** as before June,
4. despite clear instructions, ~5 members said in private that they don't understand GitHub and they've given up,
5. Eric and two expert members are satisfied with the expert list on Gaggle.email, so they abandoned `Discussion`,
6. only 7 members follow `Discussion`, a drop from 44 members on Gaggle.email,
7. more time required to write the script that will archive `Discussion` automatically,
8. GitHub not accessible in some countries (c.f. poor reputation of GitHub because of issues unrelated to cosmology),
9. more time required to write a script that will update the website to a host, (running on a local machine that is 'ON' all the time?)
10. paying a website host again,
11. more time is required to create an app for a forum on the website itself.

Clearly, ACG on GitHub doesn't work as expected and things are much worse - ACG is not flying in the right direction.  (The only positive is that the website is rendered nicely on a phone...)

It's a good time to re-evaluate the situation before you invest more time into this.  Keep in mind the [*Plan Continuation Bias*](https://en.wikipedia.org/wiki/Sunk_cost#Plan_continuation_bias) or the [*Sunk Cost*](https://en.wikipedia.org/wiki/Sunk_cost).

---
***Google search clicks/day - cosmology.info***

![cosmology-info-2023July](https://github.com/a-cosmology-group/acg/assets/16748402/98fdce41-4b30-4dda-96f5-99d715d32db0)

**Comment by ftessier** [comment]
Thank you for this hard look at the hard data. That is what science is all about! I am myself trying to argue against the _sunk cost_ fallacy at NRC, against AI tools integration in Microsoft Orifice product... sigh! 😞 

> I have exceeded my time quota for ACG.

Me too! 🤣 I agree, time is up, and the expected outcomes have not panned out:

- [x] members can write equations in Discussions,
- [ ] members modify the webpage and comment in a Git environment,
- [x] a nicer looking webpage,
- [ ] saves the cost of the web host and forum
- [ ] saves maintenance time, because members can edit the website,
- [ ] a script can be written to archive Discussions.

It was worth a try, we certainly gave it our best shot (especially you!), thanks you for putting so much effort into this "chemin de travers".

The website looks much better and is easier to maintain with git, although that is your call really since you maintain it. The better look and feel I would argue is objective. 😄 I can continue to help with git if you choose to continue development there and publish on a separate host. Updating content on a third party host is no different from any other ways to edit the website, github is then just your file store.

The sore point, apart from the unexpected blocked access 🤯  appears to be the _Discussions_. To be frank, I don't understand why, since people can still comment and reply by email, just like the mailing list. Maybe I am missing something? Perhaps members resist creating a github account?

**Reply by RedshiftDrift** [reply]
> The sore point... appears to be the Discussions. To be frank, I don't understand why, since people can still comment and reply by email, just like the mailing list. Maybe I am missing something? Perhaps members resist creating a github account?

I built the `gaggle.email` list over a few years by inviting most people, one by one - this didn't happen on their own initiative.  From the feedback I received, I can think of a few reasons why this is not working out as well as expected.

1. *"Members resist creating a GitHub account"* is only part of the problem.
2. Some only know how to `open an email`, `click a link` and `click reply`!
3. Some don't understand how the internet works and are not able to setup an account on GitHub.
4. Some find the jargon **repo**, **push**, **watch**, etc. too complicated and give up.
5. Many have never written code, so engaging on a website for coders discourage them.
6. Many are only interested in promoting their own ideas, so they don't bother reading instructions.
7. Some people don't know how to read - someone told me they had ***not*** noticed the [*Welcome to A Cosmology Group - Click* *HERE* *First*](https://github.com/orgs/a-cosmology-group/discussions) banner at the top of the `Discussion`!
8. Reluctance to post 'dissident' ideas on a public forum (anxiety, fear of being identified, anyone can steal their ideas, etc.)

Losing 20% of the members for each of these reasons, we end up with a small group...

This site is good thanks to your hard work too.  The summer season is usually slower (I'll be away for some time as well), let's see what happens in the fall.  Maybe ACG will attract a different kind of crowd with new ideas.

**Comment by ftessier** [comment]
Is [GitLab](https://gitlab.com) banned in other countries? Pushing content to another git host with Pages functionality would be a quick way to fix that problem, just push the repo there (not sure about url though). GitLab has no discussions, so that solves that _discussion_. 😆

**Reply by RedshiftDrift** [reply]
After another message, it is not clear why the Russian member is unable to create an account!  He has access to the webpage and can read `Discussion`.  It might just be an example of case `3.` above... 🤷 😕 ❔

**Reply by ftessier** [reply]
From cursory reading on this subject, it seems that _creating an account_ on github might be the operation that is blocked, while browsing github.com is still possible.

**Comment by ExpEarth** [comment]
Hi Louis and fTessier.

I agree with the points you have made on why the new format has not worked well, but I think you also missed a key point.  I had cautioned in an earlier message that there would likely be problems.  The old format worked so nicely because there was a critical mass of bright cosmology types participating in the discussions, with a wide diversity of views.  The group had a nice size, so that there were enough daily messages and at the same time not a deluge.  I described it as an ongoing cosmology discussion conducted out of Louis's living room, 24/7.  One never knew what was going to come up next for discussion.  It was quite interesting and exciting.  If all the group had migrated over to the new format and understood it completely, then some degree of success might have followed, perhaps fulfilling your goals of migration.  Even then, the 'feel' of the website might not have been as personal as before.  But when we saw that the new site was really a computer code site, and loaded with jargon (I still don't know what a pull request is, for example), and there wasn't a clear roadmap to using the new site, then it became doubtful that most of the group would follow.  In essence our 'critical mass' was lost.  A few of us tried to keep discussions going, but there was not the same energy.  Our messages started to dwindle.  At least that is how it seemed to me.

And by the way, sorry for being the one to miss the red discussion banner and "click here" entirely!  Once I found  the Discussions, it seemed like 'mission accomplished' already and no need to click anywhere.  Wrong!

The options would now seem to be to keep this one going and see how it develops or try yet another format.  Perhaps ACG could exist as a subgroup of Reddit, for instance.  Or just a continuous group email exchange.  I don't know what's best.  I do know that we are at a turning point in cosmology and we still very much need ACG or its equivalent and it's hard to imagine this without Louis still being at the centre of it.  So Louis, I hope you will take a break and focus hard on your work, but that you will be eventually return to us!

Have a good summer all !

Matt

**Comment by budrap00** [comment]
Hi Louis,

First of all, thank you for your hard work on this. It seems a little early to judge the results fairly. You are wise to step back and give the situation some time to develop. It takes time for changes like this to take full effect. It also takes time to develop an understanding of what works and doesn't work. 

From the comments you've elicited I think the main takeaway at this point would be to simplify the format. ACG's primary function is as a discussion group for people critical of the standard model. To that end, if it is at all possible, 

1. extract the discussion page from the coder's environment it is embedded in
2. the default setting for all viewers should be cosmology discussions only. Admin discussions should only be seen by those who want/need to see them
3. put a direct link to the discussions on the cosmology.info page
4. Take the summer off and come at it refreshed in September

Regards,
Bud

**Comment by ftessier** [comment]
Thank you for taking the time to pen down this postmortem. In hindsight, it seems that a key mistake (!) was holding admin discussions within this forum, making it seems more complicated than it is. My perspective was that there is little difference: a website, and a discussion forum (relayed by email, with search, threads, etc.). The only operational difference being that one has to create an account. For all the reasons you mentioned, I realize that I was wrong.

Putting a direct link to the _Discussions_ prominently on the website is a good idea. We tried reddit for EGSnrc, but to me it was _more_ confusing; we all have challenges with some interface or another I guess 😄. If reddit is already used by most people in this forum, then perhaps this can be tried. Or simply revert to an email list server such as Gaggle for discussions. 

Ultimately, reading your comments, I trust that a _single_ platform with a forum for discussions integrated in cosmology.info would work. I don't have time to try this at the moment, but intend to do so eventually. Another option would be a simple, very "flat" discussion board. [Discourse](https://discourse.org/features) offers a good example, but that particular one is 25$/month.

Finally, one way to channel attention to cosmology.info would be via the Cosmology Calculators: I guess that many people want to use those as the BB spinning top is wobbling... I would extract them from the "Code" section, put a link on the home page, and ensure that they show up in search engine, etc. Updating the visual style would also go a long way, since no such calculator that I can find out there look good by today's expectations.

**Comment by ftessier** [comment]
I am surprised by the following [metric](https://github.com/a-cosmology-group/acg/graphs/community):

![image](https://github.com/a-cosmology-group/acg/assets/1444241/aa353ced-44ca-46d9-8f2a-025856f13456)

I mean, these hundred views per week cannot all be from me! 😆

**Reply by RedshiftDrift** [reply]
If 8 logged-in users look at discussions (and replies and comments) 6+ times/day in a 6 day period, that adds up to 300 views.  This metric probably counts every visit, even if it is from the same IP.  That would explain those high numbers.
The ~16 views/day from anonymous is good though.

**Comment by khuramonline** [comment]
Equations can be written in any convenient math editor and then can be used in any forum as images.

**Comment by RedshiftDrift** [comment]
Thanks for all the comments.  It is not clear if GitHub is not accessible from most other countries.  As @ftessier said, it may be that creating a GitHub account is blocked to some people.  So far only Виталий Петриков has complained, and I am working with him to resolve the issue.

Since most comments here are relevant to the `new ACG website`, see my reply [there at https://github.com/orgs/a-cosmology-group/discussions/15](https://github.com/orgs/a-cosmology-group/discussions/15).
